docs/source/docs/configuration.md

```diff
@@ -2,19 +2,592 @@
+    extract_levels: [0, 1, 2]
+  loss:
+    image:
+      name: "lncc" # other options include "lncc", "ssd" and "gmi", for local normalised cross correlation,
+      weight: 0.1
+```
+
+The following are the default image losses. Additional arguments should be added at the
+same indent level:
+
+- `lncc`: Calls a local normalized cross-correlation type loss. Requires the following
+  arguments:
+
+  - `kernel_size`: int, optional, default=9. Kernel size or kernel sigma for
+    kernel_type="gaussian".
+  - `kernel_type`: str, optional, default="rectangular". One of "rectangular",
+    "triangular" or "gaussian"
+
+- `ssd`: Calls a sum of squared differences loss. No additional arguments required.
+
+- `gmi`: Calls a global mutual informatin loss. Requires the following arguments:
+  - `num_bins`: int, optional, default=23. Number of bins for intensity.
+  - `sigma_ratio`: float, optional, default=0.5. A hyper parameter for the gaussian
+    function.
+
+#### Label
+
+The label loss calculates dissimilarity between labels.
+
+All pre-implemented losses can be used as multi-scale or single scale losses.
+Multi-scale losses require a kernel Additionally, all losses can be weighted, so the
+following two arguments are global to all provided losses:
+
+- `weight`: float type, weight of individual loss element in total loss function.
+- `scales`: list, or None. Optional argument. If you do not pass this argument (or pass
+  the list [0], the value `null` or an empty value pair), the loss is calculated at a
+  single scale. If you past a list of length > 1, a multi-scale loss will be used.
+  WARNING: an empty list ([]) will raise an error.
+- `kernel`: str, "gaussian" or "cauchy", default "gaussian". Optional argument. Defines
+  the kernel to use for multi-scale losses.
+
+EG.
+
+```yaml
+train:
+  method: "ddf" # One of ddf, dvf, conditional
+  backbone:
+    name: "local" # One of unet, local, global
+    num_channel_initial: 16 # Int type, number of initial channels in the network. Controls the network size.
+    extract_levels: [0, 1, 2]
+  loss:
+    label:
+      weight: 1.0
+      name: "dice" # options include "dice", "cross-entropy", "mean-squared", "generalised_dice" and "jaccard"
+      scales: [1, 2]
+```
+
+The default losses require the following arguments. Additional arguments should be added
+at the same indent level:
+
+- `dice`: Calls a Dice loss on the labels, requires the following arguments:
+
+  - `binary`: bool, default is false. If true, the tensors are thresholded at 0.5.
+  - `neg_weight`: float, default=0.0. Defines the weight of the background class (0)
+    versus foreground class, which is weighted as (1-neg_weight).
+
+- `cross-entropy`: Calls a cross-entropy loss between labels, requires the following
+  arguments:
+
+  - `binary`: bool, default is false. If true, the tensors are thresholded at 0.5.
+  - `neg_weight`: float, default=0.0. Defines the weight of the background class (0)
```

Comment:
the same here, traditionally, positive weight by default should be 1, while you can reduce it between (0,1] and increase it >1, such that pos_weight / 1 is the ratio between positive and negative classes, not pos_weight / (1-pos_weight). - similar for neg_weight - if somehow normalised, it should be 0.5 / 0.5 as default and otherwise should be treated as a bug @mathpluscode

Reply:
emmmm maybe it's the name misleading,

the neg_weight = 0, means we treat 1 as foreground, 0 as background to calculate the loss

but we can treat 0 as foreground and 1 as background to calculate another loss, neg_weight is the weight for this loss

Reply:
all this stuff comes from the generalised_dice :(

Reply:
for the redundant between doc and docstring, I will do a separate issue

Reply:
ok, makes sense - only thing we need to change is here @NMontanaBrown to something like "positive and negative classes will be treated as (1-neg_weight) and neg_weight, respectively", for now, improve this later.

Reply:
How about:
Defines the weight of the ngative (0)
    and positive (1) classes: by default, L = neg_weight*negative + (1-neg_weight)*positive

Reply:
i don't think it is correct @NMontanaBrown - as @mathpluscode said, this is really not weights between foreground and background losses, they are used to replace the 0 and 1 when calculating any loss (weights in a way i guess)

Reply:
you can also be explicit saying "neg_weight weights the foreground and background classes by replacing the labels of 1s and 0s with (1-neg_weight) and neg_weight, respectively"

Reply:
Ok thanks, that is much clearer :)